src/hypertable.c

```diff
@@ -3251,6 +3303,9 @@ ts_hypertable_osm_range_update(PG_FUNCTION_ARGS)
 					   NameStr(ht->fd.schema_name),
 					   NameStr(ht->fd.table_name)),
 				errhint("Range should be set to invalid for tiered chunk"));
+	/* take lock on hypertable, keep until end of transaction */
+	// LockRelationOid(catalog_get_table_id(ts_catalog_get(), HYPERTABLE), RowExclusiveLock);
```

Comment:
Hmm do we need this, dont we want to get ShareUpdateExclusiveLock on the chunk itself to prevent someone else messing with the chunk?

Reply:
I was thinking we do not need this or a lock on the chunk either, because for OSM chunks, we neither delete them, nor update their dimension slice id. I added a test for schema changes that propagate to the OSM chunk to catch any future issues if later on we enable more modifications of OSM chunks.